resteasy-jaxrs/src/main/java/org/jboss/resteasy/spi/ResteasyProviderFactory.java

```diff
@@ -440,7 +440,12 @@ public void insertInterceptorPrecedenceBefore(String before, String newPrecedenc
 
    public static void pushContextDataMap(Map<Class<?>, Object> map)
    {
-      contextualData.setLast(map);
+      contextualData.push(map);

```

Comment:
Just to be sure we're not missing anything here, is the scenario you fixed in SynchronousDispatcher the only one in which calling contextualData.push(..) instead of contextualData.setLast(..) makes any difference? Just want to be sure we're not introducing a bug in other scenarios (as one would probably have to call pop).